include/onnxruntime/core/framework/execution_provider.h

```diff
@@ -197,34 +197,22 @@ class IExecutionProvider {
   // TODO: temparary sulotion, need to unify the interface in EP and AllocatorManager
   void TryInsertAllocator(AllocatorPtr allocator);
 
-  // creation of a fused node is not supported in a minimal build, so any EP enabled in that scenario must support
-  // compilation via GraphViewer instances.
-#if !defined(ORT_MINIMAL_BUILD)
-  /**
-  Given a list of fused_node, return create_state/compute/release_state func for each node.
-  */
-  virtual common::Status Compile(const std::vector<onnxruntime::Node*>& fused_nodes,
-                                 std::vector<NodeComputeInfo>& node_compute_funcs);
-
-  /**
-  Given a list of fused_node, return a dll that expose functions for each node.
-  For each node, there should be three symbols:
-     Create_State_${node_name}
-     Compute_${node_name}
-     Release_State_${node_name}
-  */
-  virtual common::Status Compile(const std::vector<onnxruntime::Node*>& fused_nodes,
-                                 std::string& dll_path);
-
-#endif
-
-#if !defined(ORT_MINIMAL_BUILD) || defined(ORT_EXTENDED_MINIMAL_BUILD)
   struct FusedNodeAndGraph {
     const std::reference_wrapper<onnxruntime::Node> fused_node;
     // GraphViewer that filters the full graph to the nodes that are covered by 'node'
     const std::reference_wrapper<GraphViewer> filtered_graph;
   };
 
+#if !defined(ORT_MINIMAL_BUILD) || defined(ORT_EXTENDED_MINIMAL_BUILD)
+#ifdef USE_NUPHAR
```

Comment:
based on offline discussion, I think we should still keep this previous interface for at least one release cycle to allow EP's time to migrate to the lightweight interface.

Reply:
Also, add justification for EPs to move to this new interface.

Reply:
Let's discuss the plan on how to retire old api in the review meeting.

Reply:
https://en.cppreference.com/w/cpp/language/attributes/deprecated

Reply:
@skottmckay , the [[deprecated]] attribute won't help, as here we want to deprecate an interface, not attribute. if i add the [[deprecated]] attribute to the method, we will only got the warning in framework code because we need to invoke this legacy compile api (to support the eps that haven't migrate).